src/EventListener/RespondListener.php

```diff
@@ -28,6 +30,13 @@
         'DELETE' => Response::HTTP_NO_CONTENT,
     ];
 
+    private $resourceMetadataFactory;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory = null)
```

Comment:
Aaah I see you Inject the ResourceMetadataFactory. Thanks for your PR! It's quite a large Framework and as a first-timer got me a little confused as to what's supposed to be use when and where!

Reply:
you can merge it and it will update your PR